lambda/tmc.ml

```diff
@@ -557,7 +558,11 @@ let llets lk vk bindings body =
   ) bindings body
 
 let find_candidate = function
-  | Lfunction lfun when lfun.attr.tmc_candidate -> Some lfun
+  | Lfunction lfun when lfun.attr.tmc_candidate ->
+     if not lfun.region then
+       raise (Error (Debuginfo.Scoped_location.to_location lfun.loc,
+                     Tmc_without_region));
+     Some lfun
```

Comment:
This part makes me a little nervous. Using `lfun.region` as a measure of whether something might allocate in the parent region is dubious. Tail-calls can generally break this and `unregion` will also allow breaking this. I'd much prefer that we add a constraint at type-checking time on functions marked for trmc.

Reply:
Tails-calls to functions without `lfun.region` themselves will still allocate in the parent region

Reply:
I don't see how that is possible, can you construct an example? Surely you get `region=false` if you tail call a function that has `region=false`, otherwise this seems broken.

Reply:
As discussed: this is fine but we should probably rename this field to make it's meaning clearer, and we'll need to adjust the unregion patch to keep it accurate.